### ISSUE TITLE
Unable to view mAP while training csv dataset

### ISSUE THREAD

**Issue by larrywal-express** [fizyr/keras-retinanet#1435]
Any idea on how to view mAP, precision and recall while training csv dataset?

**Comment by AndreaPi** [comment]
These questions are best asked in the [Slack channel][https://kerasteam.slack.com/archives/C7P6JEQDU]. Anyway, you can only compute the mAP on a validation set, not on the training set. For example:

```
retinanet-train \
		--backbone resnet50 \
		--weights resnet50_coco_best_v2.1.0.h5 \
		--batch-size 4 \
		--steps 400 \
		--random-transform \
		--epochs 30 \
 		csv \
	 	/data/mytrainingset/annotations.csv \
 		/data/mytrainingset/classes.csv \
                --val-annotations /data/mytestset/annotations.csv 
```

If you also want to compute the validation loss, together with the mAP on the validation set, just add the -`-compute-val-loss` argument, i.e.

```
retinanet-train \
		--backbone resnet50 \
		--weights resnet50_coco_best_v2.1.0.h5 \
		--batch-size 4 \
		--steps 400 \
		--random-transform \
		--epochs 30 \
                --compute-val-loss\
 		csv \
	 	/data/mytrainingset/annotations.csv \
 		/data/mytrainingset/classes.csv \
                --val-annotations /data/mytestset/annotations.csv 
```
